### PR TITLE
Get kadalu-quotad working as python package

### PR DESCRIPTION
**Pull request by amarts**
This is required step for getting proper support for External
Gluster storage.

More on it @ https://kadalu.io/rfcs/0001-external-gluster-storage.html

This needs suggestion and feedback.

### CODE REVIEW

File: server/setup.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from setuptools import setup
+
+
+setup(
+    name="kadalu-quotad",
+    version="0.1.0",
+    packages=["kadalu-quotad"],
```
Comment (aravindavk) [inline server/setup.py]: `-` will cause problem for package name. While creating console scripts it imports as `import kadalu-quotad` that may be problem. Change package name and dir name to `kadalu_quotad`

**Comment by aravindavk** [comment]
Current script only crawls one brick. Needs to change this to crawl multiple local bricks

File: server/kadalu-quotad/quotad.py [hunk 2]
```diff
@@ -106,9 +109,20 @@ def start():
...
     while True:
-        crawl()
-        time.sleep(1)
-        if not first_time:
+        brick_path = os.environ["BRICK_PATH"]
+        if brick_path:
+            crawl(brick_path)
+        try:
+            with open(CONFIG_FILE) as f:
+                for brick in f.readlines():
```
Comment (amarts) [inline server/kadalu-quotad/quotad.py]: @aravindavk made the crawl based on config file for now here. Isn't this enough?

Comment (aravindavk) [reply server/kadalu-quotad/quotad.py]: Sorry I missed this part. This is good

**Comment by amarts** [comment]
I guess this still needs a sysconfig script to start kadalu_quotad. And handle pushing of the package to pypi through make release scripts.

**Comment by aravindavk** [comment]
> I guess this still needs a sysconfig script to start kadalu_quotad. And handle pushing of the package to pypi through make release scripts.

Systemd service file can be included with setup.py packaging itself. Let me search how to include non-code data files.

File: server/kadalu_quotad/quotad.py [hunk 3]
```diff
@@ -6,10 +6,14 @@
 import json
 import logging
 
+# How do we handle this library dependency while packaging ?
```
Comment (amarts) [inline server/kadalu_quotad/quotad.py]: @aravindavk @sac need suggestion on how do we handle this issue with pypi package.

Comment (aravindavk) [reply server/kadalu_quotad/quotad.py]: For now, we can copy. I will think about it.

Comment (aravindavk) [reply server/kadalu_quotad/quotad.py]: I am working on the Makefile changes to create pypi package and push to pypi repo. Let me see how to package without copy

**Comment by amarts** [comment]
I am thinking for the first version, we can just give a package which we can document to run in background, instead of systemd service.

Systemd scripts can come in as a followup.

**Comment by aravindavk** [comment]
> Systemd scripts can come in as a followup.

Sure

File: server/setup.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+setup(
+    name="kadalu_quotad",
+    version="0.1.0",
+    packages=["kadalu_quotad"],
+    include_package_data=True,
+    install_requires=[""],
+    entry_points={
+        "console_scripts": [
+            "kadalu_quotad = quotad.start:main"
```
Comment (amarts) [inline server/setup.py:12]: I am not sure if this line is valid/alright ? @aravindavk ?

I see that if we get "__main__" then we call start, can I directly map the method here?

Comment (aravindavk) [reply server/setup.py:12]: `kadalu_quotad.quotad:crawl` if crawl is the starting function when this CLI is called

Comment (aravindavk) [reply server/setup.py:12]: This can be fixed with the Makefile and Pypi changes.

**Review by aravindavk — APPROVED** [review]

**Pull request merged**